### ISSUE TITLE
[$250] Security - 2FA can't be disabled for the first time after setting it up as I get logged out

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#44289]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.1-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to https://staging.new.expensify.com/
2. Log in with a new Gmail account and validate it
3. Navigate to Account settings - Security - Two-factor authentication
4. Copy the codes
5. Click on the "Next" button
6. Scan the QR code on mobile
7. Enter the verification code
8. Click on the "Next" button
9. Click on the "Got it" button
10. Click away to dismiss the modal
11. Click on "Two-factor authentication"
12. Click on " Disable two-factor authentication"
13. Click on "Disable"
14. Log back after getting logged out
15. Disable 2FA again

## Expected Result:
User should be able to disable 2FA without any issues.

## Actual Result:
2FA can't be disabled for the first time after setting it up as I get logged out of the account. I can disable it if I log back and disable it again.

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/02b76294-7cac-40b7-a8f6-a9e371079edb

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b3010fe27cdbe0fd</li>
        <li>Upwork Job ID: 1806073910011052170</li>
        <li>Last Price Increase: 2024-07-17</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ahmedGaber93</details>

**Comment by kbecciv** [comment]
@johncschuster FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by johncschuster** [comment]
I _think_ this will fit into #wave-control because it's related to 2FA, which is a domain-level feature.

**Comment by ahmedGaber93** [comment]
Waiting on proposals

**Comment by grogou** [comment]
I can’t reproduce the original issue, but I have found a new bug.

When removing 2FA, there is no logging out as described in the original issue. However, when I try to re-enable it, the codes are empty and the QR code is not scannable. After dismissing and reopening the modal, it provides valid codes along with a scannable QR code.

Here is a video demonstrating the issue.

https://github.com/Expensify/App/assets/10712881/49442056-43e6-4abe-a23b-d2118ad91674

**Comment by ahmedGaber93** [comment]
I Also can’t reproduce.

**Comment by ahmedGaber93** [comment]
@johncschuster I Also can’t reproduce this issue and facing the new issue described here https://github.com/Expensify/App/issues/44289#issuecomment-2198056804. Should we close this? or what's next?

**Comment by ahmedGaber93** [comment]
@johncschuster What do you think about https://github.com/Expensify/App/issues/44289#issuecomment-2206610759?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by ahmedGaber93** [comment]
Not overdue, I Also can’t reproduce this issue.

**Comment by ahmedGaber93** [comment]
@johncschuster What do you think about https://github.com/Expensify/App/issues/44289#issuecomment-2206610759?

**Comment by johncschuster** [comment]
Oh! Sorry I missed that comment, @ahmedGaber93! Given the behavior described in OP of this issue isn't reproducible, I think we can close this one. 

@grogou, can you make sure the new behavior you've reported in [this comment](https://github.com/Expensify/App/issues/44289#issuecomment-2198056804) has its own issue?
